blogposts/2022/introducting-migrator-service.md

```diff
@@ -0,0 +1,119 @@
+---
+title: 'Introducing the migrator service'
+author: Eric Fritz
+authorUrl: https://eric-fritz.com
+publishDate: 2022-02-15T12:00-00:00
+tags: [blog]
+slug: introducing-migrator-service
+heroImage: https://sourcegraphstatic.com/blog/introducing-migrator-service/introducing-migrator-service.jpg
+socialImage: https://sourcegraphstatic.com/blog/introducing-migrator-service/introducing-migrator-service.jpg
+published: true
+description: "We have identified and solved a huge source of pain during upgrades for Sourcegraph administrators. Here's how."
+---
+
+We have recently identified a root cause of a very frequent and insidious pain related to database migrations during Sourcegraph version upgrades. Many users handling the installation and upgrade of their Sourcegraph instance can likely attest to seeing an error similar to the following mid-way through an upgrade from one minor release to the next.
```

Comment:
Oh, boy. If there is some loosely gathered stat I could use like "dirty databases occurred for X upgrades during Y in customer instances" that would be super great.

Or if there's anything that stands out to mention here as a particular heinous example (data loss or something similar would be catastrophically delicious).

Reply:
https://github.com/sourcegraph/customer/issues/543 -- this one comes to mind, but, I know there are more, will inquire with team for more issues. I don't think our stats track issues by anything more specific than department, and this would fall under delivery issues in our current tracking